imx93: move to use ccm_rev2 clock driver

CkovMk opened the pull request:
split from #62755

danieldegrasse commented:
@CkovMk could you address the CI build failures please? You will also need to fix the merge conflict with this PR in order for CI to pass

CkovMk committed:
Revert "drivers: mcux_ccm: add support for lpuart on imx93"

This reverts commit d963900dbd56fc3663881cf54f64b81e6dcc30fb.
Since i.MX 93 is supported by mcux_ccm_rev2, remove i.MX 93 support
from mcux_ccm driver.

Signed-off-by: Chekhov Ma <chekhov.ma@nxp.com>

CkovMk commented:
Hi, @danieldegrasse 
CI issues are gone.

CkovMk committed:
imx93: change ccm driver to "imx-ccm-rev2"

i.MX93 share similiar register layout with i.MX RT11xx. Change ccm driver
to align with i.MX RT11xx, and make it easier to enable other drivers.

Signed-off-by: Chekhov Ma <chekhov.ma@nxp.com>

CkovMk commented:
Hi @mmahadevan108 ,
manifest updated, conflict resolved.
CI mostly passed, 1 failure is not related to this pr.
Thanks

CkovMk committed:
manifest: Bump up hal_nxp revision

Bump up hal_nxp revision to contain the clock frequency array
initialization code of i.MX 93

Signed-off-by: Chekhov Ma <chekhov.ma@nxp.com>

CkovMk commented:
edit commit message fix to fix issue in ci compliance check workflow.

JiafeiPan commented on west.yml:
```diff
@@ -193,7 +193,7 @@ manifest:
         - hal
     - name: hal_nxp
-      revision: d5e5358e56542b94bc65b6483396f50ed8f3172d
+      revision: 73620197038d7ba80fb1f9abd001828b9dd6a27e
```
Change it to 78d1912dbc2b5f6e114d7fd19a31053716a3b01d?

mmahadevan108 replied:
@JiafeiPan , this hash seems to be the right one for this PR.

dleach02 reviewed (changes requested):
Need to synchronize some changes i the HAL
